numberOfProvinces.js

```diff
@@ -0,0 +1,88 @@
+/*
+n cities
+
+a - b - c
+
+i: n x n matrix
+o: total number of provinces
+
+directly or indirectly
+
+Input: isConnected = [[1,1,0],[1,1,0],[0,0,1]]
+Output: 2
+
+   0 1 2
+0 [1,1,0],
+1 [1,1,0],
+2 [0,0,1]
+
+*/
+
+const { Queue } = require('./utils/queue');
+
+const CONNECTED = 1;
+
+const getNeighbors = (node, matrix) => {
+    const currentRow = matrix[node].length;
+    const neighbors = [];
+
+    for (let neighborNode = 0; neighborNode < currentRow; neighborNode++) {
+        const isSelf = neighborNode === node;
+        const isChildren = matrix[node][neighborNode] !== CONNECTED;
+
+        if (isSelf) continue;
+        if (isChildren) continue;
+
+        neighbors.push(neighborNode);
+    }
+
+    return neighbors;
+};
+
+const markComponentAsVisited = (node, matrix, visited) => {
```

Comment:
It would be a bit better to name this parameter `startNode`. Right now, it just seems like it's some random node that has no significance or meaning.